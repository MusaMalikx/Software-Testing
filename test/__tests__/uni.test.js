/** @jest-environment jsdom */
import renderer from 'react-test-renderer'
import Search from '../../components/Search/Search'
import SearchDesktop from '../../components/Search/SearchDesktop'
import SearchMobile from '../../components/Search/SearchMobile'
import Checkout from '../../pages/checkout'
import AppShell from '../../pages/appShell'
import Subcategory from "../../pages/search";
import {render, screen} from '@testing-library/react'
import '@testing-library/jest-dom'
import style from '../utility/style.js'
import React from 'react';
import MyDocument from "../../pages/_document";


describe('Checkout',()=>{

    test('Snapshot',()=>{
      const tree = renderer.create(<Checkout />).toJSON()
      expect(tree).toMatchSnapshot();
    })

    test('Tagline Text',()=>{
      render(<Checkout/>)
      const element = screen.getByTestId('checkout-final');
      expect(element).toBeInTheDocument();
      expect(element).toHaveTextContent("Coming soon!");
    })

    test('Tagline Style',()=>{
       render(<Checkout/>)
       const element = screen.getByTestId('checkout-final');
       expect(element).toBeInTheDocument();
       expect(element).toHaveClass(style.h1)
    })

})


describe('Search',()=>{

  test('Search', () => {
    const tree = renderer.create(<Search />).toJSON()
    expect(tree).toMatchSnapshot();
    })

})

describe('Search Desktop',()=>{

  test('Search Desktop', () => {
    const tree = renderer.create(<SearchDesktop />).toJSON()
    expect(tree).toMatchSnapshot();
  })

})

describe('Search Mobile',()=>{

  test('Search Mobile', () => {
    const tree = renderer.create(<SearchMobile />).toJSON()
    expect(tree).toMatchSnapshot();
  })

})

describe('App Shell',()=>{

  test('App Shell', () => {
    const tree = renderer.create(<AppShell />).toJSON()
    expect(tree).toMatchSnapshot();
  })

})

describe('Sub Category',()=>{

  test('Sub Category', () => {
    const component = renderer.create(<Subcategory />);
  
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

})

describe('My doc',()=>{

  test('basic snapshot', () => {
    const component = renderer.create(<MyDocument />);
  
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

})



