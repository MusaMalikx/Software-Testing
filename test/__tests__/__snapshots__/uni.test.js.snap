// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Shell App Shell 1`] = `null`;

exports[`Checkout Snapshot 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <div
    className="makeStyles-main-1"
  >
    <h1
      className="MuiTypography-root MuiTypography-h4 MuiTypography-colorPrimary MuiTypography-gutterBottom"
      data-testid="checkout-final"
    >
      Coming soon!
    </h1>
  </div>
</div>
`;

exports[`Search Desktop Search Desktop 1`] = `
<form
  action="/search"
  autoComplete="off"
  className="RSFSearchForm-root-15"
  onSubmit={[Function]}
  target="_top"
>
  <div
    className="RSFSearchField-root-16"
    data-empty="on"
  >
    <div
      className="RSFSearchField-inputWrap-17"
    >
      <input
        aria-label="search-text"
        className="RSFSearchField-input-18"
        name="q"
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Search..."
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="MuiPopover-root"
    onKeyDown={[Function]}
    role="presentation"
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "fixed",
        "right": 0,
        "top": 0,
        "visibility": "hidden",
        "zIndex": 1300,
      }
    }
  >
    <div
      data-test="sentinelStart"
      tabIndex={0}
    />
    <div
      className="MuiPaper-root MuiPopover-paper RSFSearchPopover-popoverPaper-22 MuiPaper-elevation8"
      style={
        {
          "opacity": 0,
          "transform": "scale(0.75, 0.5625)",
          "visibility": "hidden",
        }
      }
      tabIndex="-1"
    >
      <div
        className="RSFLoadMask-root-25 RSFLoadMask-show-29 RSFLoadMask-transparent-27"
      >
        <div
          className="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
          role="progressbar"
          style={
            {
              "height": 40,
              "width": 40,
            }
          }
        >
          <svg
            className="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx={44}
              cy={44}
              fill="none"
              r={20.2}
              strokeWidth={3.6}
              style={{}}
            />
          </svg>
        </div>
      </div>
      <div
        className="RSFSearchSuggestions-root-23"
      />
    </div>
    <div
      data-test="sentinelEnd"
      tabIndex={0}
    />
  </div>
</form>
`;

exports[`Search Mobile Search Mobile 1`] = `
<button
  aria-label="Search"
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiIconButton-label RSFSearchButton-large-3"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root RSFSearchButton-icon-2"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
      />
    </svg>
  </span>
</button>
`;

exports[`Search Search 1`] = `
[
  <button
    aria-label="Search"
    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label RSFSearchButton-large-3"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root RSFSearchButton-icon-2"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </span>
  </button>,
  <div
    className="RSFSpacer-root-14"
  />,
]
`;
